src/EIP712.sol

```diff
@@ -0,0 +1,101 @@
+    bytes32 private immutable _cachedVersionHash;
+
+    constructor() {
+        string memory name;
+        string memory version;
+        (name, version) = _domainNameAndVersion();
+        _cachedNameHash = keccak256(bytes(name));
+        _cachedVersionHash = keccak256(bytes(version));
+    }
+
+    /// @notice Returns information about the `EIP712Domain` used to create EIP-712 compliant hashes.
+    ///
+    /// @dev Follows ERC-5267 (see https://eips.ethereum.org/EIPS/eip-5267).
+    ///
+    /// @return fields The bitmap of used fields.
+    /// @return name The value of the `EIP712Domain.name` field.
+    /// @return version The value of the `EIP712Domain.version` field.
+    /// @return chainId The value of the `EIP712Domain.chainId` field.
+    /// @return verifyingContract The value of the `EIP712Domain.verifyingContract` field.
+    /// @return salt The value of the `EIP712Domain.salt` field.
+    /// @return extensions The list of EIP numbers, that extends EIP-712 with new domain fields.
+    function eip712Domain()
+        external
+        view
+        virtual
+        returns (
+            bytes1 fields,
+            string memory name,
+            string memory version,
+            uint256 chainId,
+            address verifyingContract,
+            bytes32 salt,
+            uint256[] memory extensions
+        )
+    {
+        fields = hex"0f"; // `0b1111`.
+        (name, version) = _domainNameAndVersion();
+        chainId = block.chainid;
+        verifyingContract = address(this);
+        salt = salt; // `bytes32(0)`.
+        extensions = extensions; // `new uint256[](0)`.
+    }
+
+    /// @notice Returns the `domainSeparator` used to create EIP-712 compliant hashes.
+    /// @return The 32 bytes domain separator result.
+    function domainSeparator() public view returns (bytes32) {
+        return
+            keccak256(abi.encode(_DOMAIN_TYPEHASH, _cachedNameHash, _cachedVersionHash, block.chainid, address(this)));
+    }
+
+    /// @notice Returns the EIP-712 typed hash of the `UniswapMinimalDelegationMessage(bytes32 hash)` data structure.
+    ///
+    /// @param hash The `UniswapMinimalDelegationMessage.hash` field to hash.
+    ///
+    /// @return The resulting EIP-712 hash.
+    function _hashTypedData(bytes32 hash) internal view virtual returns (bytes32) {
+        return keccak256(abi.encodePacked("\x19\x01", domainSeparator(), _hashStruct(hash)));
+    }
+
+    /// @dev Implements hashStruct(s : 𝕊) = keccak256(typeHash || encodeData(s)).
+    /// @dev See https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct.
+    ///
+    /// @return The EIP-712 `hashStruct` result.
+    function _hashStruct(bytes32 hash) internal view virtual returns (bytes32) {
+        return keccak256(abi.encode(_MESSAGE_TYPEHASH, hash));
+    }
+
+    /// @notice Returns the domain name and version to use when creating EIP-712 signatures.
+    ///
+    /// @dev MUST be defined by the implementation.
```

Comment:
same, also don't think this is helpful

im being nit picky just cause there are just a ton of comments in these two files